ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -5146,12 +5146,13 @@ let look_for_xenctrl () =
 let init () =
   look_for_forkexec () ;
   let major, minor = look_for_xen () in
+  let major = int_of_string major and minor = int_of_string minor in
```

Comment:
Can the type hypervisor be changed instead?
```ocaml
type hypervisor =
  | Xen of int * int
  (* major, minor *)
  | Other of string

```

and detect hypervisor to match this new definition? This allows the users to be protected against parsing errors

Reply:
I wonder why we need to parse at all, can't we use Xenctrl to query the hypervisor directly?
Or is this for xen-as-an-update feature?

Reply:
Xen only ever hands you those numbers in integer form, so something somewhere did a conversion to a string

Reply:
Current code is reading from files, which needs to translate strings to ints, unfortunately

Reply:
Yeah it reads them from /sys